components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/store/SyncStore.kt

```diff
@@ -0,0 +1,24 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.fxa.store
+
+import mozilla.components.lib.state.Store
+
+/**
+ * [Store] for the global [SyncState]. This should likely be a singleton.
+ */
+class SyncStore : Store<SyncState, SyncAction>(
```

Comment:
Could you expose the middlewares property here so we can add those for performing side-effects on state changes.